vars/artifactSetVersion.groovy

```diff
@@ -70,7 +91,13 @@ import groovy.text.SimpleTemplateEngine
     /** Defines the template for the timestamp which will be part of the created version. */
     'timestampTemplate',
     /** Defines the template for the automatic version which will be created. */
-    'versioningTemplate'
+    'versioningTemplate',
+    /** Controls which protocol is used for performing push operation to remote repo.
+      * Required credentials needs to be configured ('gitSshKeyCredentialsId'/'TBD').
+      * Push is only performed in case 'commitVersion' is set to 'true'.
+      * @possibleValues 'SSH', 'HTTPS', 'NONE'
+      */
+    'gitPushMode'
```

Comment:
If this should be the trigger which is not my favorite as articulated earlier (I would rather opt for getting rid of it):

I would suggest to rename since `SSH`/`HTTPS` would also be relevant to other git operations like `clone`.

We should avoid to have numerous parameters for that. 
Maybe `gitMode`  or `gitProtocol` could be an alternative.

Reply:
> I would rather opt for getting rid of it

It is not my intention to work against a majority of team members. I personally think it is better for us not to overload parameters (no implicit assumptions, one parameters controlling several things ...). Maybe we can get more opinions here ..

> would also be relevant to other git operations

But here it is only used for the push and for nothing else. What is wrong with a best-speaking parameter name making that clear? But we can find another name, we can also switch to overloaded parameters, if the majority finds that approach cool - as already outlined above.

We can open also a bigger discussion: GIT comes with two urls (fetch, push) see [here](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes). We can introduce a `gitFetchUrl` in our config (located in the general section, git clone url does not make sense since we fetch via pipeline job, the fetch url is provided by `scm`). That url contains the protocol (`ssh`, `https`) and based on that protocol we can make the desission (use sshagent / sed the credentials into the https url). With this appraoch (which would be clearly better IMO) there is no need for having a dedicated parameter describing the protocol (since it is part of the url, which is the genuin location for setting the protocol.

Reply:
I don't want to open a big discussion here. The change is too small for that.

I just foresee in the git clones for tests another area where we would need a switch if we want to consistently support HTTP.

I will leave it up to further opions ...

Reply:
From my experience having too many parameters that could be configured inconsistently causes more troubles than having assumptions which are properly documented. Enabling or Disabling or switching logic based on the availability of a step specific parameter i.e. the credentials id seems pretty natural to me though....